File: src/main/java/marisa/action/BlazeAwayAction.java
package marisa.action;

//public class BlazeAwayAction {

import marisa.marisa;
import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.utility.WaitAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.AbstractCard.CardType;
import com.megacrit.cardcrawl.cards.CardQueueItem;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.Settings;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

public class BlazeAwayAction
    extends AbstractGameAction {

  private AbstractCard card;
  AbstractPlayer p;

  public BlazeAwayAction(AbstractCard card) {
    this.duration = Settings.ACTION_DUR_FAST;
    this.p = AbstractDungeon.player;
    if (card.type == CardType.ATTACK) {
      this.card = card;
    } else {
      this.isDone = true;
    }
    marisa.logger.info(
        "BlazeAwayAction : Initialize complete ; card : " +
            card.name
    );
  }

  public void update() {

    AbstractMonster target = AbstractDungeon.getMonsters().getRandomMonster(true);
/*
    if (target == null) {
      this.isDone = true;
      ThMod.logger.info("ManaRampageAction : done");
      return;
    }
    */
    AbstractDungeon.player.limbo.group.add(card);
    card.current_x = (Settings.WIDTH / 2.0F);
    card.current_y = (Settings.HEIGHT / 2.0F);
    card.target_x = (Settings.WIDTH / 2.0F - 300.0F * Settings.scale);
    card.target_y = (Settings.HEIGHT / 2.0F);
    card.freeToPlayOnce = true;
    card.purgeOnUse = true;
    card.targetAngle = 0.0F;
    card.drawScale = 0.12F;
    marisa.logger.info(
        "BlazeAwayAction : card : " +
            card.cardID +
            " ; target : " +
            target.id
    );

    card.applyPowers();
    AbstractDungeon.actionManager.currentAction = null;
    AbstractDungeon.actionManager.addToTop(this);
    AbstractDungeon.actionManager.cardQueue.add(
        new CardQueueItem(card, target)
    );

    if (!Settings.FAST_MODE) {
      AbstractDungeon.actionManager.addToTop(new WaitAction(Settings.ACTION_DUR_MED));
    } else {
      AbstractDungeon.actionManager.addToTop(new WaitAction(Settings.ACTION_DUR_FASTER));
    }

    this.isDone = true;
  }

}

File: src/main/kotlin/marisa/action/BlazeAwayAction.kt
package marisa.action

import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.actions.utility.WaitAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.AbstractCard.CardType
import com.megacrit.cardcrawl.cards.CardQueueItem
import com.megacrit.cardcrawl.core.Settings
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import marisa.MarisaContinued

class BlazeAwayAction(val card: AbstractCard) : AbstractGameAction() {

    init {
        duration = if (Settings.FAST_MODE) Settings.ACTION_DUR_FASTER else Settings.ACTION_DUR_MED
        isDone = card.type != CardType.ATTACK
        MarisaContinued.logger.info("BlazeAwayAction : Initialize complete ; card : ${card.name}")
    }

    override fun update() {
        val target = AbstractDungeon.getMonsters().getRandomMonster(true)

        AbstractDungeon.player.limbo.group.add(card)
        card.apply {
            current_x = Settings.WIDTH / 2.0f
            current_y = Settings.HEIGHT / 2.0f
            target_x = Settings.WIDTH / 2.0f - 300.0f * Settings.scale
            target_y = Settings.HEIGHT / 2.0f
            freeToPlayOnce = true
            purgeOnUse = true
            targetAngle = 0.0f
            drawScale = 0.12f
        }

        MarisaContinued.logger.info(
            """BlazeAwayAction : card : ${card.cardID} ; target : ${target.id}"""
        )
        card.applyPowers()
        AbstractDungeon.actionManager.run {
            currentAction = null
            addToTop(this@BlazeAwayAction)
            cardQueue.add(CardQueueItem(card, target))
            addToTop(WaitAction(duration))
        }
        isDone = true
    }
}
